sdk/communication/azure-communication-administration/src/main/java/com/azure/communication/administration/CommunicationIdentityClientBuilder.java

```diff
@@ -68,6 +73,22 @@ public CommunicationIdentityClientBuilder pipeline(HttpPipeline pipeline) {
         return this;
     }
 
+    /**
+     * Sets the {@link TokenCredential} used to authenticate HTTP requests.
+     *
+     * @param tokenCredential {@link TokenCredential} used to authenticate HTTP requests.
+     * @return The updated {@link CommunicationIdentityClientBuilder} object.
+     * @throws NullPointerException If {@code tokenCredential} is null.
+     */
+    public CommunicationIdentityClientBuilder credential(TokenCredential tokenCredential) {
+        this.tokenCredential = Objects.requireNonNull(tokenCredential, "'tokenCredential' cannot be null.");
+        if (accessKeyCredential != null) {
+            logger.info("'accessKey' is being replaced with 'credential', only one authorization method can be used.");
+            this.accessKeyCredential = null;
+        }
```

Comment:
We should not change the value of `accessKeyCredential` the user has already set. This will silently set the value to null and the user will not know about it - logs are not always noticed. I would rather do the validation at the very end when `build*Client` is called. Check if both are set and then throw an exception.